### ISSUE TITLE
Hebrew names with spaces show inconsistency between web-shown and on-disk directories

### ISSUE THREAD

**Issue by ezaton** [nextcloud/server#17056]
### Steps to reproduce
1. Using NextCloud Linux client, sync directories with long Hebrew names with spaces in them
2. Some of them (randomly) would cause the NextCloud agent to fail
3. The directory shown on the Web, as well as on the NextCloud agent would be shorter, missing some words. On disk, however, the name is correct.
4. Attempting occ files:scan and files:cleanup did not help

### Expected behaviour
Directory names should match the on-disk names

### Actual behaviour
Directory names on the web interface show a shorter, trimmed, version of the original on-disk name, which results in a loop when clicked on (sends you back to the user's 'All files' home). See screenshot of the web view of the directory, the info to the right, the 
![with_links](https://user-images.githubusercontent.com/6866518/64507817-5c094a80-d2e4-11e9-9e0b-31d74ca7ce74.png)

From CLI (screenshot):
![Screenshot from 2019-09-09 09-32-25](https://user-images.githubusercontent.com/6866518/64507976-cae6a380-d2e4-11e9-9971-85c77d55194c.png)

### Server configuration

**Operating system**:
Centos 7.7
**Web server:**
Apache 2.4.41 from CodeIT
**Database:**
MariaDB 10.4.7

**PHP version:**
7.3.9 (remi)
**Nextcloud version:** (see Nextcloud admin page)
16.0.4 stable
**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh installation

**Where did you install Nextcloud from:**
NextCloud web site

**Signing status:**
<details>
A screenshot of the browser display, vs, a screenshot of the settings (which then work) vs a screenshot of the terminal window showing the directory (Hebrew is RTL, so the text shows from left to right). 

<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
"No errors have been found."
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
Enabled:
  - accessibility: 1.2.0
  - activity: 2.9.1
  - bruteforcesettings: 1.4.0
  - cloud_federation_api: 0.2.0
  - comments: 1.6.0
  - dav: 1.9.2
  - federatedfilesharing: 1.6.0
  - federation: 1.6.0
  - files: 1.11.0
  - files_pdfviewer: 1.5.0
  - files_rightclick: 0.15.1
  - files_texteditor: 2.8.0
  - files_trashbin: 1.6.0
  - files_versions: 1.9.0
  - files_videoplayer: 1.5.0
  - firstrunwizard: 2.5.0
  - gallery: 18.3.0
  - logreader: 2.1.0
  - lookup_server_connector: 1.4.0
  - nextcloud_announcements: 1.5.0
  - notifications: 2.4.1
  - oauth2: 1.4.2
  - password_policy: 1.6.0
  - privacy: 1.0.0
  - provisioning_api: 1.6.0
  - recommendations: 0.4.0
  - serverinfo: 1.6.0
  - sharebymail: 1.6.0
  - support: 1.0.0
  - survey_client: 1.4.0
  - systemtags: 1.6.0
  - theming: 1.7.0
  - twofactor_backupcodes: 1.5.0
  - updatenotification: 1.6.0
  - viewer: 1.1.0
  - workflowengine: 1.6.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - files_sharing
  - user_ldap
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.barnoy.co.il"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "16.0.4.1",
        "overwrite.cli.url": "https:\/\/nextcloud.barnoy.co.il",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false
    },
    "apps": {
        "accessibility": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": ""
        },
        "activity": {
            "enabled": "yes",
            "installed_version": "2.9.1",
            "types": "filesystem"
        },
        "backgroundjob": {
            "lastjob": "7909"
        },
        "bruteforcesettings": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": ""
        },
        "cloud_federation_api": {
            "enabled": "yes",
            "installed_version": "0.2.0",
            "types": "filesystem"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "logging"
        },
        "core": {
            "backgroundjobs_mode": "cron",
            "installed.bundles": "[\"CoreBundle\"]",
            "installedat": "1567503277.799",
            "lastcron": "1568009101",
            "lastupdateResult": "[]",
            "lastupdatedat": "1568009234",
            "oc.integritycheck.checker": "[]",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "theming.variables": "db81cddf52fdb3c8ca1e4c859e214124",
            "vendor": "nextcloud"
        },
        "dav": {
            "buildCalendarSearchIndex": "yes",
            "enabled": "yes",
            "installed_version": "1.9.2",
            "regeneratedBirthdayCalendarsForYearFix": "yes",
            "types": "filesystem"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "enabled": "yes",
            "installed_version": "1.11.0",
            "types": "filesystem"
        },
        "files_external": {
            "enabled": "no",
            "installed_version": "1.7.0",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": ""
        },
        "files_rightclick": {
            "enabled": "yes",
            "installed_version": "0.15.1",
            "types": ""
        },
        "files_sharing": {
            "enabled": "no",
            "installed_version": "1.8.0",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "yes",
            "installed_version": "2.8.0",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "filesystem,dav"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "filesystem,dav"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "yes",
            "installed_version": "2.5.0",
            "types": "logging"
        },
        "gallery": {
            "enabled": "yes",
            "installed_version": "18.3.0",
            "types": ""
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.1.0",
            "types": ""
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": "authentication"
        },
        "nextcloud_announcements": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "pub_date": "Mon, 02 Sep 2019 00:00:00 +0200",
            "types": "logging"
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "2.4.1",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.4.2",
            "types": "authentication"
        },
        "password_policy": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": ""
        },
        "privacy": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "prevent_group_restriction"
        },
        "recommendations": {
            "enabled": "yes",
            "installed_version": "0.4.0",
            "types": ""
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "filesystem"
        },
        "support": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "survey_client": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "last_report": "{\"id\":\"oc3vs6mflp83\",\"items\":[[\"server\",\"version\",\"16.0.4.1\"],[\"server\",\"code\",\"other\"],[\"server\",\"enable_avatars\",\"yes\"],[\"server\",\"enable_previews\",\"yes\"],[\"server\",\"memcache.local\",\"none\"],[\"server\",\"memcache.distributed\",\"none\"],[\"server\",\"asset-pipeline.enabled\",\"no\"],[\"server\",\"filelocking.enabled\",\"yes\"],[\"server\",\"memcache.locking\",\"none\"],[\"server\",\"debug\",\"no\"],[\"server\",\"cron\",\"cron\"],[\"php\",\"version\",\"7.3.9\"],[\"php\",\"memory_limit\",536870912],[\"php\",\"max_execution_time\",0],[\"php\",\"upload_max_filesize\",2097152],[\"database\",\"type\",\"mysql\"],[\"database\",\"version\",\"10.4.7\"],[\"database\",\"size\",11378688],[\"apps\",\"accessibility\",\"1.2.0\"],[\"apps\",\"activity\",\"2.9.1\"],[\"apps\",\"bruteforcesettings\",\"1.4.0\"],[\"apps\",\"cloud_federation_api\",\"0.2.0\"],[\"apps\",\"comments\",\"1.6.0\"],[\"apps\",\"dav\",\"1.9.2\"],[\"apps\",\"federatedfilesharing\",\"1.6.0\"],[\"apps\",\"federation\",\"1.6.0\"],[\"apps\",\"files\",\"1.11.0\"],[\"apps\",\"files_pdfviewer\",\"1.5.0\"],[\"apps\",\"files_rightclick\",\"0.15.1\"],[\"apps\",\"files_sharing\",\"1.8.0\"],[\"apps\",\"files_texteditor\",\"2.8.0\"],[\"apps\",\"files_trashbin\",\"1.6.0\"],[\"apps\",\"files_versions\",\"1.9.0\"],[\"apps\",\"files_videoplayer\",\"1.5.0\"],[\"apps\",\"firstrunwizard\",\"2.5.0\"],[\"apps\",\"gallery\",\"18.3.0\"],[\"apps\",\"logreader\",\"2.1.0\"],[\"apps\",\"lookup_server_connector\",\"1.4.0\"],[\"apps\",\"nextcloud_announcements\",\"1.5.0\"],[\"apps\",\"notifications\",\"2.4.1\"],[\"apps\",\"oauth2\",\"1.4.2\"],[\"apps\",\"password_policy\",\"1.6.0\"],[\"apps\",\"privacy\",\"1.0.0\"],[\"apps\",\"provisioning_api\",\"1.6.0\"],[\"apps\",\"recommendations\",\"0.4.0\"],[\"apps\",\"serverinfo\",\"1.6.0\"],[\"apps\",\"sharebymail\",\"1.6.0\"],[\"apps\",\"support\",\"1.0.0\"],[\"apps\",\"survey_client\",\"1.4.0\"],[\"apps\",\"systemtags\",\"1.6.0\"],[\"apps\",\"theming\",\"1.7.0\"],[\"apps\",\"twofactor_backupcodes\",\"1.5.0\"],[\"apps\",\"updatenotification\",\"1.6.0\"],[\"apps\",\"viewer\",\"1.1.0\"],[\"apps\",\"workflowengine\",\"1.6.0\"],[\"stats\",\"num_files\",10713],[\"stats\",\"num_users\",2],[\"stats\",\"num_storages\",4],[\"stats\",\"num_storages_local\",1],[\"stats\",\"num_storages_home\",3],[\"stats\",\"num_storages_other\",0],[\"stats\",\"num_comments\",0],[\"stats\",\"num_comment_markers\",0],[\"stats\",\"num_systemtags\",0],[\"stats\",\"num_systemtags_mappings\",0],[\"files_sharing\",\"num_shares\",0],[\"files_sharing\",\"num_shares_user\",0],[\"files_sharing\",\"num_shares_groups\",0],[\"files_sharing\",\"num_shares_link\",0],[\"files_sharing\",\"num_shares_link_no_password\",0],[\"files_sharing\",\"num_fed_shares_sent\",0],[\"files_sharing\",\"num_fed_shares_received\",0],[\"encryption\",\"enabled\",\"no\"],[\"encryption\",\"default_module\",\"no\"]]}",
            "last_sent": "1567527602",
            "types": ""
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "logging"
        },
        "theming": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "types": "logging"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": ""
        },
        "updatenotification": {
            "enabled": "yes",
            "files_rightclick": "0.15.1",
            "installed_version": "1.6.0",
            "types": "",
            "update_check_errors": "0"
        },
        "viewer": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": ""
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "filesystem"
        }
    }
}

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No
#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
[Mon Sep 09 09:00:50.538494 2019] [cgid:error] [pid 78094:tid 140344658175744] [client 31.206.14.115:60363] AH01264: script not found or unable to stat: /var/www/cgi-bin/ViewLog.asp
[Mon Sep 09 09:07:37.822380 2019] [authz_core:error] [pid 78094:tid 140344926512896] [client 84.108.166.157:34544] AH01630: client denied by server configuration: /var/www/html/nextcloud/data/.ocdata

</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
[nextcloud.log](https://github.com/nextcloud/server/files/3588857/nextcloud.log)

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by ezaton** [comment]
Update: It has nothing to do with directory and spaces. It incorrectly identifies (by PHP, I assume, as part of the PHP code protection) some Hebrew characters and ignores them. See the attached screenshots of two files, all of which are in the sync root, and all have an incorrect name compared to their info. Also note how the sync agent modified quotes mark (") to %22, and created two copies of the directory.
![Screenshot from 2019-09-11 09-40-49](https://user-images.githubusercontent.com/6866518/64674133-a28fae00-d478-11e9-9281-76972d9144d3.png)
![Screenshot from 2019-09-11 09-40-22](https://user-images.githubusercontent.com/6866518/64674134-a3284480-d478-11e9-8019-999adecd5b6e.png)

**Comment by ezaton** [comment]
I think this is related to issue [#14981](https://github.com/nextcloud/server/issues/14981). Seems to follow similar pattern.
The main problem here is that some of the data on the computer was lost due to file name changes in the sync process.

**Comment by ezaton** [comment]
An update. This is in the level of PHP/WebDav. Look at the file called "1". The actual name of the file is "הראל 1", and the Hebrew (first word, because Hebrew is read from right to left) is omitted and a leading space remains. The file path appears in the DB correctly, but the name is incorrect in oc_filecache:
  62538 |       3 | files/Desktop-PC/Documents/הראל 1.pdf                                                                                                                                                                                                                                                 | 01c96d8437adceecf1c8cfc55ed297ab |    346 |  1.pdf 
I have removed the entry and rescanned, and the file appears correctly in the web interface now. 
I have performed a drop of the oc_filecache table, and recreated it, and files seem to look correct now.

Post-mortem - it seems that the NextCloud agent for Linux has created faulty entries in the oc_filecache table, and these entries were not corrected, but remained, until manually removed. 
NextCloud was updated, during that time to version 16.0.5, which might be related (cached entries were not refreshed - that I have tested, but now they are being created correctly). 
Anyhow - since this bug has not been assigned almost a month now, I believe I should close it. Hopefully - someone will make use of this info, and maybe, if I encounter another issues, I will open/reopen a more focused bug.
